### PR TITLE
Add Unit Testing via Check

### PR DESCRIPTION
**Pull request by EdHurtig**
This PR adds the support for unit testing using [libcheck](https://libcheck.github.io/check/), a unit testing framework for C projects.

**Summary**

- Compile `core/` into `libcore.a` as well as an executable `core`.  This is required for the `check` framework to build a test binary that tests the functionality defined in core.
   - Eventually core should become libcore, and a wrapper executable that just has a `main()` should be used.
- Tests are defined under the `tests/check/` folder. An initial test `check_states.c` contains an initial test for the state machine.
    - TODO: Port the old state machine tests from July into this file
- Add `ctest` to the Jenkinsfile test stages.

To compile and run tests.

```
cd ./proj
cmake <options> ..
make clean all install
ctest --verbose
```

In summary: CTest is the runner that executes test binaries, Check is the testing library that compiles the test binaries.

### CODE REVIEW

**Review by QuantumAlmonds — COMMENTED** [review]
Why does main just return and not do anything?

**Comment by EdHurtig** [comment]
It is a stub: https://libcheck.github.io/check/doc/check_html/check_3.html

**Review by QuantumAlmonds — APPROVED** [review]
Got it thanks!

**Comment by EdHurtig** [comment]
In the future, as this test_states.c file actually accumulates a sufficient number of tests, a suite will be defined like this https://github.com/libcheck/check/blob/master/doc/example/tests/check_money.c
